src/ConnectKafka.h

```diff
@@ -13,15 +13,54 @@ class ConnectKafka : public ConnectKafkaParentClass {
 
 public:
   ConnectKafka(std::string Broker, std::string ErrStr);
+  ~ConnectKafka() {
+    if (Consumer) {
+      Consumer->close();
+      /*
+       * Wait for RdKafka to decommission.
+       * This is not strictly needed (with check outq_len() above), but
+       * allows RdKafka to clean up all its resources before the application
+       * exits so that memory profilers such as valgrind wont complain about
+       * memory leaks.
+       */
+      RdKafka::wait_destroyed(5000);
+    }
+  }
+
   std::unique_ptr<RdKafka::Metadata> queryMetadata() override;
+
   std::string GetAllTopics() override;
+
   void SubscribeToTopic(const std::vector<std::string> &Topic) override;
+
   bool CheckIfTopicExists(std::string Topic) override;
-  virtual std::string Consume(std::string Topic) override;
-  virtual std::vector<int32_t> GetTopicPartitions(std::string Topic) override;
+
+  virtual std::pair<std::string, bool>
```

Comment:
If you mark with `override` then the `virtual` keyword is redundant, it can be removed. (same for other member functions of this class)